### PR TITLE
Feature/reset date

### LINKED ISSUE #408: Add reset date
Currently reputation on `TOTAL` reputation mode counts all existing reputations.

To facilitate some kind of easy custom reputation counting, a date could be defined which will then define from which date reputation should be counted. This could be also easily added to the views and would not require any additional changes to the reputation profile or rank retrieval itself.

- [x] Add a new command group called `reset_date` to `repadmin`
  - [x] Add command to define a reset date
  - [x] Add command to remove the reset date
- [x] Integrate the reset date in the total ranking view

### PR DESCRIPTION
**Pull request by RainbowDashLabs**
**Checklist**
- [x] I made changes to commands
  - [x] I fully localised the command
  - [x] I fully checked the commands functionality

- [x] I added new localisation codes
  - [x] I fully translated it for all languages
  - [x] I sorted properties alphabetically

- [x] I made changes to the database
  - [x] I added my changed to a patch file
  - [x] I made sure my database was up to date
  - [x] I checked that the migration works

- [x] I made changes to internal structure 

**Short Description**
Adds a reset date which allows to ignore reputation received before this date.

**Detailed Description**

Reputation given before a defined date is ignored. This is accomplished by modifying the total view. A reset date is not supported for reputation modes other then `total`.

Closes #408

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -15,7 +15,7 @@ repositories {
 
 dependencies {
     //discord
-    implementation("de.chojo", "cjda-util", "2.7.3+alpha.20") {
+    implementation("de.chojo", "cjda-util", "2.7.4+alpha.20-SNAPSHOT") {
```
Comment (Taucher2003) [inline build.gradle.kts:18]: I think  we want to switch to a stable version again before merging (or at least before releasing the next update)

File: src/main/java/de/chojo/repbot/commands/repadmin/handler/resetdate/RemoveResetDate.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import de.chojo.jdautil.interactions.slash.structure.handler.SlashHandler;
+import de.chojo.jdautil.wrapper.EventContext;
+import de.chojo.repbot.dao.provider.Guilds;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+public class RemoveResetDate implements SlashHandler {
+    private final Guilds guilds;
+
+    public RemoveResetDate(Guilds guilds) {
+
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/repadmin/handler/resetdate/RemoveResetDate.java:12]:
```suggestion
```

File: src/main/java/de/chojo/repbot/commands/repadmin/handler/resetdate/SetResetDate.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        this.guilds = guilds;
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event, EventContext context) {
+        var year = event.getOption("year").getAsInt();
+        var month = event.getOption("month").getAsInt();
+        var day = event.getOption("day").getAsInt();
+
+        var date = LocalDate.of(year, month, day);
```
Comment (Taucher2003) [inline src/main/java/de/chojo/repbot/commands/repadmin/handler/resetdate/SetResetDate.java:26]: We probably want to guard against bad combinations (for example: `2022-02-30`)

File: src/main/resources/database/postgresql/1/patch_23.sql [hunk 4]
```diff
@@ -1,3 +1,4 @@
+--feature/guild-cleanup
```
Comment (Taucher2003) [inline src/main/resources/database/postgresql/1/patch_23.sql:1]:
```suggestion
-- feature/guild-cleanup
```

File: src/main/resources/locale_de.properties [hunk 5]
```diff
@@ -172,6 +172,17 @@ command.repadmin.reputation.set.amount.description=Der zu setzende Betrag.
 command.repadmin.reputation.set.description=Die Reputation eines Nutzers festlegen.
 command.repadmin.reputation.set.message.set=Setzt die Reputation eines Nutzers von %USER% auf %VALUE%.
 command.repadmin.reputation.set.user.description=Der Nutzer, der die Reputation festlegt.
+command.repadmin.resetdate.current.description=Siehe das aktuelle Rücksetzdatum.
```
Comment (Taucher2003) [inline src/main/resources/locale_de.properties:175]:
```suggestion
command.repadmin.resetdate.current.description=Zeigt das aktuelle Rücksetzdatum.
```

File: src/main/resources/locale_de.properties [hunk 6]
```diff
@@ -172,6 +172,17 @@ command.repadmin.reputation.set.amount.description=Der zu setzende Betrag.
...
+command.repadmin.resetdate.current.message.notset=Kein Rücksetzdatum definiert.
+command.repadmin.resetdate.current.message.set=Rücksetzdatum auf %DATE% gesetzt.
+command.repadmin.resetdate.description=Verwalte das Rücksetzdatum der Reputation.
+command.repadmin.resetdate.remove.description=Das Rücksetzdatum entfernen
```
Comment (Taucher2003) [inline src/main/resources/locale_de.properties:179]:
```suggestion
command.repadmin.resetdate.remove.description=Das Rücksetzdatum entfernen.
```

File: src/main/resources/locale_de.properties [hunk 7]
```diff
@@ -172,6 +172,17 @@ command.repadmin.reputation.set.amount.description=Der zu setzende Betrag.
...
+command.repadmin.resetdate.remove.message.removed=Rücksetzdatum entfernt.
+command.repadmin.resetdate.set.day.description=Tag
+command.repadmin.resetdate.set.description=Ruf ausblenden, der vor einem bestimmten Datum eingegangen ist
```
Comment (Taucher2003) [inline src/main/resources/locale_de.properties:182]:
```suggestion
command.repadmin.resetdate.set.description=Reputation ausblenden, welche vor einem bestimmten Datum eingegangen ist.
```

File: build.gradle.kts (hunk 1 above)
Comment (RainbowDashLabs) [reply build.gradle.kts:18]: The review of it is assigned to you as well :P

Comment (RainbowDashLabs) [reply build.gradle.kts:18]: Oh you reviewed it already. Its merged and renovate should open a PR soon.

**Review by Taucher2003 — APPROVED** [review]

**Pull request merged**